### PR TITLE
Fix 183

### LINKED ISSUE #183: Solve problem with {} tags in identifier
### Description
In the indicator class, we use the identifier attribute as the unique identifier of a given class. Since this was also used as the output variable name for the created netCDF, we introduced formatting tags in the identifier attribute. The problem is that the same identifier is also used in other contacts (finch) to identify uniquely the process, but before the class is called, we can't format the identifier string. 

We should have two distinct attributes: the unique class identifier, and the output variable name. The identifier should not contain {}, but the output variable name could. 

More generally, we should clarify which attributes need to be clearly defined without arguments tags, and which can include tags.

### PR DESCRIPTION
**Pull request by huard**
* **Please check if the PR fulfills these requirements**
- [x] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, etc.)
Indicator class has a `var_name` attribute used to name the output variable. The identifier identifies the function, so has no {} tags. 

* **Does this PR introduce a breaking change?** (Has there been an API change?)
Likely yes. This PR needs more testing before merging. 

Fixes #183

### CODE REVIEW

File: xclim/atmos/_precip.py [hunk 1]
```diff
@@ -48,7 +49,8 @@ class PrTas(Indicator2D):
                                     compute=indices.max_n_day_precipitation_amount,
                                     )
 
-wetdays = Pr(identifier='r{thresh}mm',
+wetdays = Pr(identifier='wetdays',
+             var_name='r{thresh}mm',
```
Comment (tlogan2000) [inline xclim/atmos/_precip.py]: ```atmos.wetdays(pr, thresh='10 mm/day')``` 
gives 'r10 mm/daymm' as an output variable name

Comment (tlogan2000) [reply xclim/atmos/_precip.py]: We would want to somehow extract the value of the pint unit here I think

File: xclim/atmos/_precip.py [hunk 2]
```diff
@@ -48,7 +49,8 @@ class PrTas(Indicator2D):
...
              units='days',
              standard_name='number_of_days_with_lwe_thickness_of_precipitation_amount_at_or_above_threshold',
              long_name='Number of Wet Days (precip >= {thresh} mm)',
```
Comment (tlogan2000) [inline xclim/atmos/_precip.py]: same idea here ... this is giving 'Number of Wet Days (precip >= **10 mm/daymm**)

**Comment by huard** [comment]
@tlogan2000 should be good now.

**Pull request merged**